lazyscribe/repository.py

```diff
@@ -183,9 +183,12 @@ def load_artifact(
             The artifact object.
         """
         # Search for the artifact
-        artifact = self._search_artifact_versions(
-            name=name, version=version, match=match
-        )
+        try:
+            artifact = self._search_artifact_versions(
+                name=name, version=version, match=match
+            )
+        except ValueError as err:
+            raise ArtifactLoadError("No valid artifact found") from err
```

Comment:
I don't agree with this change, personally. The `ValueError` raised by `Repository._search_artifact_versions` is an accurate description of the exception.

Reply:
Ok, I'll repush